internal/operator/controller.go

```diff
@@ -330,6 +344,67 @@ func (c *Controller) reconcile(o *stanv1alpha1.NatsStreamingCluster) error {
 	return nil
 }
 
+func (c *Controller) reconcilePodTemplate(o *stanv1alpha1.NatsStreamingCluster) error {
+	pods, err := c.findRunningPods(o.Name, o.Namespace)
+	if err != nil {
+		return err
+	}
+
+	desiredImage := o.Spec.Image
+	if desiredImage == "" {
+		desiredImage = DefaultNATSStreamingImage
+	}
+
+	var desiredAnnotations map[string]string
+	podTemplate := o.Spec.PodTemplate
+	if podTemplate != nil {
+		desiredAnnotations = podTemplate.GetObjectMeta().GetAnnotations()
+	}
+
+	for _, pod := range pods {
+		currentImage := pod.Spec.Containers[0].Image
+		currentAnnotations := pod.GetObjectMeta().GetAnnotations()
+		delete(currentAnnotations, "kubernetes.io/psp") // Ignore k8s auto-applied annotations
```

Comment:
@wallyqs This header is auto-assigned by k8s, so it doesn't come from the PodTemplate. I wonder if there is a more generic way to do this in order to compare with the annotations in the actual pod evenly. For example, should I just look at all `kuernetes.io/*` keys and remove them, or is this enough? At the least, this works in my k8s cluster on the IBM Cloud platform.